src/main/java/com/ably/kafka/connect/ChannelConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.ably.kafka.connect;
+
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.ChannelMode;
+import io.ably.lib.types.ChannelOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CHANNEL_CONFIG;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_CIPHER_KEY;
+import static com.ably.kafka.connect.ChannelSinkConnectorConfig.CLIENT_CHANNEL_PARAMS;
+
+interface ChannelConfig {
+    String getName();
+
+    ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException;
+}
+
+class ChannelConfigImpl implements ChannelConfig {
+    private final ChannelSinkConnectorConfig sinkConnectorConfig;
+
+    public ChannelConfigImpl(ChannelSinkConnectorConfig sinkConnectorConfig) {
+        this.sinkConnectorConfig = sinkConnectorConfig;
+    }
+
+    @Override
+    public String getName() {
+        return sinkConnectorConfig.getString(CHANNEL_CONFIG);
+    }
+
+    public ChannelOptions getOptions() throws ChannelSinkConnectorConfig.ConfigException {
+        final Logger logger = LoggerFactory.getLogger(ChannelSinkConnectorConfig.class);
+        ChannelOptions opts;
+        String cipherKey = sinkConnectorConfig.getString(CLIENT_CHANNEL_CIPHER_KEY);
+
+        if (cipherKey != null && !cipherKey.trim().isEmpty()) {
+            try {
+                opts = ChannelOptions.withCipherKey(cipherKey);
+            } catch (AblyException e) {
+                logger.error("Error configuring channel cipher key", e);
```

Comment:
I'm not sure if there is an established convention in this codebase (I've not looked) but I would generally avoid reporting exceptions in more than one place as this can be confusing for users. I would just throw the exception and leave it at that.

Reply:
Unfortunately there is not an established convention as it looks like the logging and exception throwing was happening on multiple places. I moved the ones here to be dealt with  / logged in task class https://github.com/ably/kafka-connect-ably/pull/50/commits/78ac0790786a859c13dde2447e1f63e3c8c4043a